### ISSUE TITLE
Floating point exception(core dumped)

### ISSUE THREAD

**Issue by Sugercanee** [VAST-AI-Research/UniRig#49]
When I run **bash launch/inference/generate_skin.sh --input results/dog_skeleton.fbx --output results/dog_skin.fbx**

Info: Deleted 1 data-block(s)
Now processing results/dog_skeleton.fbx...
Now processing results/dog_skeleton.fbx...
FBX version: 7400
save to: tmp/results/dog_skeleton
100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:01<00:00,  1.67s/it]
1 models processed
done
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:47: FutureWarning: `torch.cuda.amp.custom_fwd(args...)` is deprecated. Please use `torch.amp.custom_fwd(args..., device_type='cuda')` instead.
  _TORCH_CUSTOM_FWD = amp.custom_fwd(cast_inputs=torch.float16)
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:96: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:162: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:242: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:331: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:368: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:388: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/spconv/pytorch/functional.py:411: FutureWarning: `torch.cuda.amp.custom_bwd(args...)` is deprecated. Please use `torch.amp.custom_bwd(args..., device_type='cuda')` instead.
  @_TORCH_CUSTOM_BWD
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/timm/models/layers/__init__.py:48: FutureWarning: Importing from timm.models.layers is deprecated, please import via timm.layers
  warnings.warn(f"Importing from {__name__} is deprecated, please import via timm.layers", FutureWarning)
Seed set to 12345
load task config: configs/task/quick_inference_unirig_skin.yaml
load data config: configs/data/quick_inference.yaml
load transform config: configs/transform/inference_skin_transform.yaml
load model config: configs/model/unirig_skin.yaml
WARNING: use BatchNorm in ptv3obj !!!
load system config: configs/system/skin.yaml
not found: /data1/cxz/Pretrained_models/UniRig/models--VAST-AI--UniRig/snapshots/15d305165ef4448ed85e7bdf8f100d3ca7b0123e/skin/articulation-xl/model.ckpt
💡 Tip: For seamless cloud uploads and versioning, try installing [litmodels](https://pypi.org/project/litmodels/) to enable LitModelCheckpoint, which syncs automatically with the Lightning model registry.
GPU available: True (cuda), used: True
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
Restoring states from the checkpoint path at /data1/cxz/Pretrained_models/UniRig/models--VAST-AI--UniRig/snapshots/15d305165ef4448ed85e7bdf8f100d3ca7b0123e/skin/articulation-xl/model.ckpt
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/lightning/pytorch/trainer/call.py:282: Be aware that when using `ckpt_path`, callbacks used to create the checkpoint need to be provided during `Trainer` instantiation. Please add the following callbacks: ["ModelCheckpoint{'monitor': 'val_loss_sum', 'mode': 'min', 'every_n_train_steps': 0, 'every_n_epochs': 1, 'train_time_interval': None}"].
LOCAL_RANK: 0 - CUDA_VISIBLE_DEVICES: [1]
Loaded model weights from the checkpoint at /data1/cxz/Pretrained_models/UniRig/models--VAST-AI--UniRig/snapshots/15d305165ef4448ed85e7bdf8f100d3ca7b0123e/skin/articulation-xl/model.ckpt
/home/cxz/anaconda3/envs/UniRig/lib/python3.11/site-packages/lightning/pytorch/trainer/connectors/data_connector.py:425: The 'predict_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=111` in the `DataLoader` to improve performance.
Predicting DataLoader 0:   0%|                                                                                                   | 0/1 [00:00<?, ?it/s]/home/cxz/anaconda3/envs/UniRig/lib/python3.11/contextlib.py:105: FutureWarning: `torch.backends.cuda.sdp_kernel()` is deprecated. In the future, this context manager will be removed. Please see `torch.nn.attention.sdpa_kernel()` for the new context manager, with updated signature.
  self.gen = func(*args, **kwds)
/home/cxz/Code_field/UniRig/src/model/unirig_skin.py:359: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  'offset': torch.tensor(batch['offset']),
launch/inference/generate_skin.sh: line 80: **1027225 Floating point exception(core dumped)** python run.py --task=configs/task/quick_inference_unirig_skin.yaml --seed=12345 --input=results/dog_skeleton.fbx --output=results/dog_skin.fbx --npz_dir=tmp --data_name=raw_data.npz
done

**Comment by theshackdev** [comment]
**Fixed — the crash is old spconv (2.3.6); upgrade to the cu126 package line: `spconv-cu126==2.3.8`.**

Hit this exact crash (`generate_skin.sh` → `Floating point exception (core dumped)` at the first `SubMConv3d` in the PTv3 backbone) with `spconv-cu120==2.3.6` on torch 2.7.0+cu128 / Python 3.11. It reproduced identically on an RTX 5060 Ti (sm_120), an RTX 4070 (sm_89) and an RTX 5070 Ti Mobile, with every conv algo (`SPCONV_ALGO=native` included) — so it is not GPU-arch related.

The trap: the `spconv-cu120` PyPI package line dead-ends at **2.3.6 (Apr 2023)**. The Dec 2024 releases (2.3.7/2.3.8) ship only as `spconv-cu121` / `spconv-cu124` / `spconv-cu126`, so if you're on the cu120 line you never see them:

```bash
pip uninstall -y spconv-cu120 cumm-cu120
pip install spconv-cu126==2.3.8    # pulls cumm-cu126 0.7.11
```

The cu126 wheel coexists fine with a cu128 torch in the same process (spconv bundles its own CUDA and doesn't link torch's). After the swap, a minimal `SubMConv3d` repro passes on all three GPUs and the full `generate_skin.sh` runs clean end-to-end — output FBX has every vertex skin-weighted.

Note: we did **not** need the `numpy<2` downgrade suggested in traveller59/spconv#725 — numpy 2.4.4 works fine with spconv 2.3.8 + cumm 0.7.11.

Environment: Fedora 44, driver 595.80, torch 2.7.0+cu128, Python 3.11, flash_attn 2.7.4.post1.

*(Fix was tracked down with the help of Claude Fable 5.)*
